lib/Commands/AddClient.php

```diff
@@ -77,10 +77,18 @@ protected function execute(InputInterface $input, OutputInterface $output) {
 		if (!\is_bool($allowSubDomains)) {
 			throw new \InvalidArgumentException('Please enter true or false for allowed-sub-domains.');
 		}
+		try {
+			$this->clientMapper->findByName($name);
```

Comment:
@jvillafanez this is how our mappers are designed. They throw exception when nothing found...

Reply:
My thought process here is:
1. Find the client -> I guess we should store the client we've found, why isn't there an assignment?
2. Write a message -> ok
3. Return an error -> ???
4. Catch an exception -> comment says it's good, ok.

Just by adding a comment:
1. Find the client -> I guess we should store the client we've found, why isn't there an assignment?
2. Comment explaining we expect not to find the client and we expect the exception -> ok, so what is inside the "try" block is the "exception" code and we can jump over it under normal conditions.
3. Check the exception -> this is fine and we don't need to do anything.

In the second case is easier to understand what you're trying to do without stopping, otherwise I think you'll have to stop reading the code to understand what you're trying to do.

Reply:
thanks for clarification.